presto-verifier/src/main/java/com/facebook/presto/verifier/event/VerifierQueryEvent.java

```diff
@@ -0,0 +1,312 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.verifier.event;
+
+import io.airlift.event.client.EventField;
+import io.airlift.event.client.EventType;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+@EventType("VerifierQuery")
+public class VerifierQueryEvent
+{
+    private final String suite;
+    private final String testId;
+    private final String name;
+
+    private final boolean succeeded;
+    private final boolean skipped;
```

Comment:
I would slightly prefer the way it is. Enum may make the Java code look nice, but they are eventually converted to string in the json output, and its `one-of` property gets lost. Clients see a string field and needs to process it based on some hard-coded values, whereas if boolean is used, clients only need to know the field name, since its value is binary.

Reply:
At the same time that leads us to think of situations, can both `succeeded` and `skipped` be true and then check the different fields. Having a single value that states clearly the situation is better - you just need to read a single field and then these values are familiar enough for someone to infer.